### PR TITLE
Michael Agard

### PR DESCRIPTION
**Pull request by fellanonymous**
@bradiowave

### CODE REVIEW

File: basic-javascript/project-2.js [hunk 1]
```diff
@@ -68,27 +76,33 @@ const isPrime = num => {
...
+    return arr;
   }
-  return arr;
 };
 
 const addItemToArray = (arr, item) => {
-  arr.push(item);
-  return arr;
+  if (arr === undefined) {
+    return null;
```
Comment (bradencodes) [inline basic-javascript/project-2.js:101]: You should check if the type of arr is actually an array. If you make `arr` a string, it is still defined and it will try to `.push` the item onto it.

File: basic-javascript/project-1.spec.js [hunk 2]
```diff
@@ -1,3 +1,429 @@
...
+    })
+    it('return "invalid entry" if nothing was entered', () => {
+      const expected = "invalid entry";
+      const actual = helpers.areSameLength();
+      expect(actual).toEqual(expected)
+    })
+    it('return "invalid entry" if a string was entered', () => {
+      const expected = "invalid entry";
+      const actual = helpers.areSameLength("5");
+      expect(actual).toEqual(expected)
```
Comment (bradencodes) [inline basic-javascript/project-1.spec.js:52]: Be more specific and say that this will fail when only *one* string was entered.

File: basic-javascript/project-1.spec.js [hunk 3]
```diff
@@ -1,3 +1,429 @@
...
+    })
+    it('return "invalid entry" if a string was entered', () => {
+      const expected = "invalid entry";
+      const actual = helpers.areEqual("5");
+      expect(actual).toEqual(expected)
+    })
+  })
+  describe('lessThanNinety()', () => {
+    it('checks the value of two numbers and returns if they are equal', () => {
```
Comment (bradencodes) [inline basic-javascript/project-1.spec.js:73]: Make sure to change the description of what the test does when copy-pasting ;)

File: basic-javascript/project-1.spec.js [hunk 4]
```diff
@@ -1,3 +1,429 @@
...
+    it('return "invalid entry" if a string was entered', () => {
+      const expected = "invalid entry";
+      const actual = helpers.lessThanNinety("5");
+      expect(actual).toEqual(expected)
+    })
+  })
+  describe('greaterThanFifty()', () => {
+    it('checks the value of two numbers and returns if they are equal', () => {
+      const expected = true;
+      const actual = helpers.greaterThanFifty(51);
```
Comment (bradencodes) [inline basic-javascript/project-1.spec.js:92]: Good job testing the number right where it should be true.

File: basic-javascript/project-2.spec.js [hunk 5]
```diff
@@ -1,3 +1,121 @@
...
+    it('should return "invalid entry" if a string was entered', () => {
+      expect(helpers.getBiggest("5")).toEqual("invalid entry");
+    });
+  });
+  describe('isTenOrFive()', () => {
+    it('should return true if it was 10 or 5"', () => {
+      expect(helpers.isTenOrFive(5)).toEqual(true);
+    });
+    it('should return false if nothing was entered', () => {
+      expect(helpers.isTenOrFive()).toEqual(false);
```
Comment (bradencodes) [inline basic-javascript/project-2.spec.js:19]: Tests for cases where you give the function a number but expect the function to return false.

**Comment by bradencodes** [comment]
## MVP
- [x] Write tests for project-1.js
- [x] Write tests for project-2.js

## Stretch
- [x] Correct bugs and refactor the code
- [ ] Write tests for arrays.js
- [ ] Write tests for object.js

## What Looks Great
- MVP nearly complete! Nice work
- Frequent and descriptive commits
- Code is well formatted and easy to read
- Titles for tests are descriptive
- Tests are thorough and check edge cases and valid data.

## What Could Be Improved
- Finish up the last few tests in project-2

## What Could Be Added
- Write tests where the user provides valid data, but the function will still fail.

## Rating (1-3)
**3**
